src/lib.rs

```diff
@@ -1115,8 +1121,20 @@ fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
 
+/// Returns whether the given MAKEFLAGS indicate that there is an available
+/// jobserver that uses a named pipe (fifo)
+fn uses_named_pipe_jobserver(makeflags: &OsStr) -> bool {
+    makeflags
+        .to_str()
+        // auth option as defined in
+        // https://www.gnu.org/software/make/manual/html_node/POSIX-Jobserver.html#POSIX-Jobserver
+        .map(|flags| flags.contains("--jobserver-auth=fifo:"))
+        .unwrap_or(false)
```

Comment:
IMO `to_string_lossy` is better here, since the path could contain non utf-8 characters

```suggestion
    makeflags
        .to_string_lossy()
        // auth option as defined in
        // https://www.gnu.org/software/make/manual/html_node/POSIX-Jobserver.html#POSIX-Jobserver
        .contains("--jobserver-auth=fifo:")
```